umbraco-cloud/set-up/project-settings/umbraco-cicd/samplecicdpipeline/github-actions-pwsh.md

```diff
@@ -0,0 +1,145 @@
+- From the root folder
+  - `cloud.zipignore`
+- From `powershell` folder
+  - `Get-LatestDeployment.ps1`
+  - `Get-ChangesById.ps1`
+  - `New-Deployment.ps1`
+  - `Add-DeploymentPackage.ps1`
+  - `Start-Deployment.ps1`
+  - `Test-DeploymentStatus.ps1`
+- From the `powershell/github` folder
+  - `main.yml`
+  - `cloud-sync.yml` 
+  - `cloud-deployment.yml`
+
+{% endhint %}
+
+## Import Cloud project repository to GitHub
+Go to your repositories in GitHub and click on "New".
+
+- Create a new empty repository in GitHub, and note down the clone URL.
+- Go to the Umbraco Cloud Portal and clone your cloud project down locally. [This article](../../../working-locally.md#cloning-an-umbraco-cloud-project) describes how you can find the clone URL.
+- Now working locally remove the Git Remote called `origin`, which points to Umbraco Cloud
+
+ ```sh 
+ git remote remove origin
+ ```
+
+- Optionally rename branch `master` to `main`
+
+ ```sh 
+ # optional step
+ git branch -m  main
+ git symbolic-ref HEAD refs/heads/main
+ ```
+
+- Add a new remote called origin and pointing to the GitHub clone URL
+
+ ```sh 
+ git remote add origin https://github.com/{your-organization}/{your-repository}.git
+ ```
+
+- Push local repository to Github
+
+Now we can move on to setting up a pipeline.
+
+## Set up Github repository variables
+
+The pipeline needs to know which Umbraco Cloud project to deploy to. In order to do this you will need the `Project ID` and the `API Key`. [This article](README.md#obtaining-the-project-id-and-api-key) describes how to get those values.
+
+- Create a new repository in GitHub
+- Now go to the repository in GitHub, and click on the Settings section.
+- Expand secrets and variables in the left-hand menu titled Security and click on Actions.
+
+<figure><img src="../../../../.gitbook/assets/image (6).png" alt=""><figcaption><p>Security and Actions menu GitHub</p></figcaption></figure>
+
+- Create a `repository secret` called `UMBRACO_CLOUD_API_KEY` with the `API Key` value from the Umbraco Portal.
+- Create a `repository variable` with the name `PROJECT_ID` and the value `Project ID` value from the Umbraco Portal.
+
+{% hint style="info" %}
+If you want to use other names for the variable or secret, you need to rename the environment variables in `main.yml`.
+
+```yaml
+env:
+  projectId: ${{ vars.PROJECT_ID }} # change the part inside the curly braces
+  umbracoCloudApiKey: ${{ secrets.UMBRACO_CLOUD_API_KEY }} # change the part inside the curly braces
+```
+
+{% endhint %}
+
+Now Github is set up with all the needed information to be able to deploy your Cloud project back to Umbraco Cloud.
+
+## Set up the GitHub Actions pipeline
+
+While working with the project on you local machine, do the following to prepare the pipeline.
+- Copy the `cloud.zipignore` file to the root of your repository
+- Make a copy of the `.gitignore` and call it `cloud.gitignore`
+  - Both files should be in the root of your repository
```

Comment:
note for Jesper so he remembers:
add a line about gitignore the git-patch.diff file non cloud .gitignore